src/shared/condition.c

```diff
@@ -196,6 +199,8 @@ typedef enum {
 
 static OrderOperator parse_order(const char **s) {
         static const char *const prefix[_ORDER_MAX] = {
+                [ORDER_FNMATCH_EQUAL] = "=$",
+                [ORDER_FNMATCH_UNEQUAL] = "!=$",
```

Comment:
This adds the two new operators to every settting that users `parse_order`: ConditionKernelVersion, ConditionOSRelease, ConditionMemory, ConditionCPUs. I don't think this is appropriate.

Please add a flag here that specifies whether the two new operators are allowed, and only pass `true` in some callers.

I think ConditionOSRelease and ConditionKernelVersion *could* be extended to allow this too. But it should be a separate commit, and the documentation must be updated to match.

Reply:
I'm going to change this.

> I think ConditionOSRelease and ConditionKernelVersion could be extended to allow this too. But it should be a separate commit, and the documentation must be updated to match.

Is it OK for you to do this in a following pull request?